docs/src/main/asciidoc/sending-emails.adoc

```diff
@@ -0,0 +1,227 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+
+include::./attributes.adoc[]
+= {project-name} - Sending emails
+
+This guide demonstrates how your {project-name} application can send emails using an SMTP server.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* The SMTP hostname, port and credentials, and an email address
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+* GraalVM installed if you want to run in native mode.
+
+== Architecture
+
+In this guide, we are going to see how you can send emails from a {project-name} application.
+It covers simple emails, attachments, inlined attachments, the reactive and imperative APIs...
+
+== Creating the Maven Project
+
+Create a new project with the following command:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=sending-email-quickstart \
+    -Dextensions="mailer"
+----
+
+If you already have an existing project, add the `mailer` extension:
+
+[source]
+----
+mvn quarkus:add-extensions -Dextensions="mailer"
+----
+
+== Configuring the mailer
+
+The {project-name} mailer is using SMTP. In the `src/main/resources/application.properties` file, you need to configure the host, port, username, password as well as the other configuration aspect.
+Note that the password can also be configured using system properties and environment variables.
+
+Here is an example using _sendgrid_:
+
+[source]
+----
+quarkus.mailer.from=test@quarkus.io
+quarkus.mailer.host=smtp.sendgrid.net
+quarkus.mailer.port=465
+quarkus.mailer.ssl=true
+quarkus.mailer.username=....
+quarkus.mailer.password=....
+----
+
+The following table lists the properties that can be configured:
+
+[cols=*,options="header"]
+|===
+
+| Property | Type | Comment
+
+| from | String | the default _from_ address, without, all emails will need an explicit value
+| mock | Boolean | if set to `true`, the emails are not sent, just printed on the console.
+| bounceAddress | String | the default bounce address
+| host | String | the SMTP server host name
+| port | Integer | the SMTP server port
+| username | String | the username
+| password | String | the password
+| ssl | Boolean | Enables or disables SSL
+| trustAll | Boolean | Set whether to trust all certificates
+| maxPoolSize | Integer | Set the maximum pool size
```

Comment:
Pool of what?

Reply:
Doesn't matter as long as `olympic` is an accepted value

Reply:
It's the number of connection to the server.

Reply:
I've rewritten the documentation and updated the javadoc to be clearer.